### PR TITLE
UPSTREAM Don't pretty print by default

### PR DESCRIPTION
**Pull request by derekwaynecarr**
cc @smarterclayton 

https://github.com/GoogleCloudPlatform/kubernetes/pull/9361

### CODE REVIEW

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -271,32 +271,49 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
...
+
+func isPrettyPrint(req *http.Request) bool {
+	pp := req.URL.Query().Get("pretty")
+	if len(pp) > 0 {
+		pretty, _ := strconv.ParseBool(pp)
+		return pretty
+	}
+	userAgent := req.UserAgent()
+	// This covers basic all browers and cli http tools
+	if strings.HasPrefix(userAgent, "curl") || strings.HasPrefix(userAgent, "Wget") || strings.HasPrefix(userAgent, "Mozilla/5.0") {
```
Comment (smarterclayton) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/apiserver.go]: Really?

**Comment by derekwaynecarr** [comment]
[test]

**Comment by derekwaynecarr** [comment]
@smarterclayton - I think we want this perf fix.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**